azure/windows-webapp/windows-webapp/main.tf

```
@@ -0,0 +1,371 @@
+      use_32_bit_worker      = lookup(site_config.value, "use_32_bit_worker", null)
+      local_mysql_enabled    = lookup(site_config.value, "local_mysql_enabled", null)
+      always_on              = lookup(site_config.value, "always_on", false)
+      vnet_route_all_enabled = lookup(site_config.value, "vnet_route_all_enabled", null)
+      load_balancing_mode    = lookup(site_config.value, "load_balancing_mode", null)
+
+      minimum_tls_version     = "1.2"
+      scm_minimum_tls_version = "1.2"
+      http2_enabled           = lookup(site_config.value, "http2_enabled", false)
+      websockets_enabled      = lookup(site_config.value, "websockets_enabled", false)
+      default_documents       = var.default_documents
+
+      api_definition_url    = lookup(site_config.value, "api_definition_url", null)
+      api_management_api_id = lookup(site_config.value, "api_management_api_id", null)
+
+      health_check_path                 = lookup(site_config.value, "health_check_path", null)
+      health_check_eviction_time_in_min = lookup(site_config.value, "health_check_eviction_time_in_min", null)
+
+      container_registry_use_managed_identity       = lookup(site_config.value, "container_registry_use_managed_identity", true)
+      container_registry_managed_identity_client_id = lookup(site_config.value, "container_registry_managed_identity_client_id", null)
+      remote_debugging_enabled                      = lookup(site_config.value, "remote_debugging_enabled", false)
+      remote_debugging_version                      = lookup(site_config.value, "remote_debugging_version", null)
+
+      dynamic "ip_restriction" {
+        for_each = local.access_rules
+        content {
+          name     = ip_restriction.name
+          priority = ip_restriction.priority
+          action   = ip_restriction.action
+
+          ip_address                = ip_restriction.ip_address
+          service_tag               = ip_restriction.service_tag
+          virtual_network_subnet_id = ip_restriction.virtual_network_subnet_id
+          headers                   = ip_restriction.headers
+        }
+      }
+      dynamic "scm_ip_restriction" {
+        for_each = local.access_rules
+        content {
+          name     = scm_ip_restriction.name
+          priority = scm_ip_restriction.priority
+          action   = scm_ip_restriction.action
+
+          ip_address                = scm_ip_restriction.ip_address
+          service_tag               = scm_ip_restriction.service_tag
+          virtual_network_subnet_id = scm_ip_restriction.virtual_network_subnet_id
+          headers                   = scm_ip_restriction.headers
+        }
+      }
+
+      dynamic "cors" {
+        for_each = var.cors == null ? [] : [1]
+        content {
+          allowed_origins     = lookup(var.cors, "allowed_origins", null)
+          support_credentials = lookup(var.cors, "support_credentials", null)
```

lookup is not needed here as the optional type is defined in variables, and omitting the default value by setting it to null is depreciated. 
```suggestion
          support_credentials = var.cors["support_credentials"]
```

---

This is specifically referenced in Terraform documentation:

https://developer.hashicorp.com/terraform/language/functions/lookup

```
For historical reasons, the default parameter is actually optional. However, omitting default is deprecated since v0.7 because that would then be equivalent to the native index syntax, map[key].
```

---

In the case of using Lookup, the default behavior if a value is not found was to error out and the original implementation of optional values didn't quite work the same, explaining this. Can factor this change in though for both app types.

---

This has been raised under https://github.com/orgs/ans-coe/projects/1/views/1?pane=issue&itemId=26669612 - ideally tackle this in the same PR.